src/screens/contacts/contacts.js

```diff
@@ -0,0 +1,106 @@
+class ContactsScreen extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            allUsers: [],
+           
+            listUsers: []
+        }
+    }
+
+    async componentDidMount() {
+        let user = await _getUser()
+        this.convertUserObjectToArray(this.props.usersProfile, user);
+    }
+
+    convertUserObjectToArray = (usersObj = {}, own) => {
+        let usersArr = [];
+        Object.keys(usersObj).forEach((user = {}) => {
+            if (own !== user) {
+                usersArr.push({ ...usersObj[user], key: user });
+            }
+        });
+        this.setState({ allUsers: usersArr, listUsers: usersArr, mainUser: own });
+    }
+
+    listItem = ({ item }) => {
+     
+        if (!item) {
+            return null
+        }
+        return (
+            <ListItem key={item.name} bottomDivider>
+                <Avatar source={item.avatar} />
+                <ListItem.Content>
+                    <ListItem.Title>{item.name + ' (' + item.gender + ')'}</ListItem.Title>
+                    <ListItem.Subtitle>{item.dob}</ListItem.Subtitle>
+                    <View style={styles.subtitleView}>
+                        {/* <Image source={require('../images/rating.png')} style={styles.ratingImage}/> */}
+                        <Text style={styles.ratingText}>{item.location}</Text>
+                    </View>
+                </ListItem.Content>
+                
+            </ListItem>
+        )
+    }
+
+    render() {
+        const { usersContacts = {} } = this.props;
+        const { allUsers, mainUser } = this.state;
+        let contactsArr = usersContacts[mainUser];
+        let showContacts = allUsers.filter((user) => {
+            if(contactsArr && contactsArr.includes(user.name)){
+                return user;
+            }
+        })
+        return (
+            <SafeAreaView
+                style={styles.homeScreenContainer}
+            >
+                <Text style={styles.heading}>Contacts</Text>
+              
+                <View style={{ flex: 1, }}>
+                    {showContacts && showContacts.length ?
+                        <FlatList
+                            style={{ width: '100%' }}
+                            renderItem={this.listItem}
+                            data={showContacts}
+                            keyExtractor={user => JSON.stringify(user)}
+                        />
+                        :
+                        <View style={{ flex: 1, alignItems: "center", justifyContent: "center" }}>
+                            <Text>No Data Found</Text>
+                        </View>
+                    }
+                </View>
+            </SafeAreaView >
+        )
+    }
+}
+
+const mapStateToProps = ({ profileReducer: usersProfile, homeReducer: { usersContacts } }) => (
+    {
+        ...usersProfile,
+        usersContacts
+    }
+);
+
+const mapDispatchToProps = dispatch => bindActionCreators({
```

Comment:
You can also use a different syntax for mapDispatchToProps. This syntax is a very simplified one.
Like this:
const mapDispatchToProps = {
  sortDrivers,
  searchDrivers,
};
You can explore this syntax more else I can discuss with you separately.

Reply:
Sure will discuss this